### ISSUE TITLE
excluding vios hipers

### ISSUE THREAD

**Issue by smurphyit** [IBM/ansible-power-aix#377]
When running a flrtvc report only against a host, it is finding both HIPER/SEC for both AIX levels and vios levels.  See example below.  It would be nice to specify (or have system determine if it's a VIOS or not), then only show those levels.  Example run:

    "result": {
        "changed": false,
        "failed": false,
        "meta": {
            "0.report": [
                "Fileset|Current Version|Type|EFix Installed|Abstract|Unsafe Versions|APARs|Bulletin URL|Download URL|CVSS Base Score|Reboot Required|Last Update|Fixed In",
                "bos.rte.control|7.2.5.201|sec||NOT FIXED - AIX is vulnerable to a denial of service due to libxml2|7.2.5.200-7.2.5.201|IJ47597 / CVE-2023-28484 / CVE-2023-29469|https://aix.software.ibm.com/aix/efixes/security/libxml2_advisory5.asc|https://aix.software.ibm.com/aix/efixes/security/libxml2_fix5.tar|CVE-2023-28484:5.5 CVE-2023-29469:5.5|NO|07/25/2023|7200-05-07",
                "bos.rte.control|7.2.5.201|sec||NOT FIXED - AIX is vulnerable to a denial of service due to libxml2|7.2.5.200-7.2.5.201|IJ47597 / CVE-2023-28484 / CVE-2023-29469|https://aix.software.ibm.com/aix/efixes/security/libxml2_advisory5.asc|https://aix.software.ibm.com/aix/efixes/security/libxml2_fix5.tar|CVE-2023-28484:5.5 CVE-2023-29469:5.5|NO|07/25/2023|3.1.4.30",

As you can see it has the "Fixed In:" for AIX 7200-05-07 and then another iteration for "3.1.4.30" which would be the IOSlevel if it was a VIOS, which this system is not.

Ideally, the VIOS output would only be provided if the target was a VIOS, otherwise, just the AIX vulnerabilities should be listed.

**Comment by nitismis** [comment]
Thanks for reporting.

**Comment by schamola** [comment]
Hi @smurphyit

We looked into this issue, and have found the root cause. It is being caused because in the FLRTVC script, there is no functionality for getting the system type. It just shows all the available fixes based on the fixes that are already installed on the machine regardless of the OS (VIOS or AIX) it is running on. Hence, fixes for both AIX and its VIOS counterpart are being listed.
That said, we have made some changes in our code and will be pushing them soon.
